nanoFramework.Azure.Devices.Client/DeviceClient.cs

```diff
@@ -160,9 +160,17 @@ public bool Open()
             if (!string.IsNullOrEmpty(ModelId))
             {
                 userName += $"&model-id={HttpUtility.UrlEncode(ModelId)}";
-
             }
 
+            // compose product info
+            StringBuilder productInfo = new StringBuilder();
```

Comment:
For performance reasons, we've seen with the json class that string concatenations is faster.
So maybe just use $"" all the time?